internal/annotators/http/pki_test.go

```diff
@@ -0,0 +1,177 @@
+			anno, err := pki.Do(tt.ctx, data)
+			test.CheckError(err, tt.expectError, tt.name, t)
+			if err == nil {
+				result, err := annotators.VerifySignature(tt.cfg.Signature.PublicKey, anno)
+				if err != nil {
+					t.Error(err.Error())
+				} else if !result {
+					t.Error("signature not verified")
+				}
+			}
+		})
+	}
+}
+
+func buildRequest(keys config.SignatureInfo) (*http.Request, []byte, error) {
+	type sample struct {
+		Key   string `json:"key"`
+		Value string `json:"value"`
+	}
+
+	t := sample{Key: "keyA", Value: "This is some test data"}
+	b, _ := json.Marshal(t)
+
+	req := httptest.NewRequest("POST", "/foo?param=value&foo=bar&baz=batman", bytes.NewReader(b))
+	req.Header.Set("Host", "example.com")
+
+	ticks := time.Now()
+	now := ticks.String()
+	req.Header.Set("Date", now)
+	req.Header.Set(contentType, string(contracts.ContentTypeJSON))
+	req.Header.Set(contentLength, strconv.FormatInt(req.ContentLength, 10))
+
+	fields := []string{string(method), string(path), string(authority), contentType, contentLength}
+	headerValue, signature, err := signRequest(ticks, fields, keys, req)
+
+	req.Header.Set("Signature-Input", headerValue)
+	req.Header.Set("Signature", signature)
+
+	return req, b, err
+}
+
+func signRequest(ticks time.Time, fields []string, keys config.SignatureInfo, req *http.Request) (string, string, error) {
+	headerValue := "" //This will be the value returned for populating the Signature-Input header
+	inputValue := ""  //This will be the value used as input for the signature
+	sigParams := ""   //This will be used to build the signatureParams field
+
+	for i, f := range fields {
+		headerValue += fmt.Sprintf("\"%s\"", f)
+		switch f {
+		case contentType:
+			inputValue += fmt.Sprintf("\"%s\" %s", f, req.Header.Get(contentType))
+		case contentLength:
+			inputValue += fmt.Sprintf("\"%s\" %s", f, strconv.FormatInt(req.ContentLength, 10))
+		case string(method):
+			inputValue += fmt.Sprintf("\"%s\" %s", f, req.Method)
+		case string(authority):
+			inputValue += fmt.Sprintf("\"%s\" %s", f, req.Host)
+		case string(scheme):
+			scheme := strings.ToLower(strings.Split(req.Proto, "/")[0])
+			inputValue += fmt.Sprintf("\"%s\" %s", f, scheme)
+		case string(requestTarget):
+			inputValue += fmt.Sprintf("\"%s\" %s", f, req.RequestURI)
+		case string(path):
+			inputValue += fmt.Sprintf("\"%s\" %s", f, req.URL.Path)
+		case string(query):
+			var query string = "?" + req.URL.RawQuery
+			inputValue += fmt.Sprintf("\"%s\" %s", f, query)
+		case string(queryParams):
+			queryParamsRawMap := req.URL.Query()
+			var queryParams []string
+			for key, value := range queryParamsRawMap {
+				b := new(bytes.Buffer)
+				fmt.Fprintf(b, ";name=\"%s\": %s", key, value[0])
+				queryParams = append(queryParams, b.String())
+			}
+
+			inputValue += fmt.Sprintf("\"%s\" %s", f, query)
+		}
+
+		sigParams += fmt.Sprintf("\"%s\"", f)
+		inputValue += "\n"
+		if i < len(fields)-1 {
+			headerValue += " "
+			sigParams += " "
+		}
+	}
+
```

Comment:
I don't see the `alg` or `key` fields being included as part of the Signature-Input string. See the following:

https://github.com/tsconn23/alvarium-sdk-go/blob/aff0f18286da4d1b3a5a6a12e07ab5e47e186b20/internal/annotators/http/pki_test.go#L147